fanpy/analysis/dataframe/geminal.py

```
@@ -91,6 +91,14 @@ def read_csv(self, filename, **kwargs):
         self.wfn_ref_sd = self.metadata["wfn_ref_sd"]
         self._index_view = self.metadata["index_view"]
 
+        self.dataframe = self.dataframe.reset_index()
+        if self.metadata["index_view"] == "geminals":
+            self.dataframe = self.dataframe.set_index(["geminal", "pair"])
+        elif self.metadata["index_view"] == "determinants":
+            self.dataframe = self.dataframe.set_index(["geminal", "determinants"])
+        elif self.metadata["index_view"] == "formatted determinants":
+            self.dataframe = self.dataframe.set_index(["geminal", "formatted determinants"])
+
```

do we need to raise an error if `self.metadata["index_view"]` is not one of `"geminals", "determinants", "formatted determinants"`? E.g. having an `else` section to catch invalid values:
``` python
if self.metadata["index_view"] == "geminals":
    self.dataframe = self.dataframe.set_index(["geminal", "pair"])
elif self.metadata["index_view"] == "determinants":
    self.dataframe = self.dataframe.set_index(["geminal", "determinants"])
elif self.metadata["index_view"] == "formatted determinants":
    self.dataframe = self.dataframe.set_index(["geminal", "formatted determinants"])
else:
    raise ValueError("Index view has to be...")
```

---

That's a good idea. I also found that the ```.read_csv``` can be improved to handle any cases missing wfn information from JSON file. I will submit commits about that soon.